util/realistic_data/daily_ingestor.py

```diff
@@ -28,35 +28,83 @@ def main():
             tzinfo=tz.gettz("Europe/London"),
         )
         for file in hdf_files
+        # Includes files only, any directories are ignored
+        if file.is_file()
     }
+
+    in_progress_date_format = f"{hdf_file_date_format}%S"
+    current_datetime = (
+        datetime.now().replace(microsecond=0).strftime(in_progress_date_format)
+    )
+    Path(f"{hdf_data_directory}/in_progress_{current_datetime}").mkdir(
+        parents=True,
+        exist_ok=True,
+    )
+    in_progress_dir = Path(f"{hdf_data_directory}/in_progress_{current_datetime}")
```

Comment:
Could this go above line 39? 
Then you can do `in_progress_dir.mkdir` on line 39 without repeating the concatenation of the file path.

Reply:
Good spot! That means `mkdir()` can go onto one line so that's nice :)